lms/djangoapps/certificates/tests/test_support_views.py

```diff
@@ -326,6 +331,33 @@ def test_regenerate_user_has_no_certificate(self):
         num_certs = GeneratedCertificate.eligible_certificates.filter(user=self.student).count()
         self.assertEqual(num_certs, 1)
 
+    def test_regenerate_user_with_invalidated_cert(self):
+        """ Verify if the certificate is marked as invalid, if so, regenerate
+        the certificate and de-activate the invalidated entry. """
+
+        # mark certificate as invalid
+        self._invalidate_certificate(self.cert)
+        self.assertTrue(self._invalidate_cert_exists())
+        # after invalidation certificate become un-available.
+        self.assertTrue(
+            self._generated_certificate_exist(
+                user=self.student, status=CertificateStatuses.unavailable
+            )
+        )
+        # Should be able to regenerate
+        response = self._regenerate(
+            course_key=self.CERT_COURSE_KEY,
+            username=self.STUDENT_USERNAME
+        )
+        self.assertEqual(response.status_code, 200)
+
+        self.assertFalse(self._invalidate_cert_exists())
+        # we'd expect that the certificate status will be "notpassing"
```

Comment:
What's the reason the regenerated certificate have status `notpassing`? Is it because of the test data setup or other reason? Can you add in comments the reason for the `notpassing` status?